clang-tools-extra/clang-tidy/bugprone/IncorrectEnableSharedFromThisCheck.h

```diff
@@ -0,0 +1,35 @@
+//===--- IncorrectEnableSharedFromThisCheck.h - clang-tidy ------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_CLANG_TOOLS_EXTRA_CLANG_TIDY_BUGPRONE_INCORRECTENABLESHAREDFROMTHISCHECK_H
+#define LLVM_CLANG_TOOLS_EXTRA_CLANG_TIDY_BUGPRONE_INCORRECTENABLESHAREDFROMTHISCHECK_H
+
+#include "../ClangTidyCheck.h"
+
+namespace clang::tidy::bugprone {
+
+/// Detects if a class or struct publicly inherits from
+/// ``std::enable_shared_from_this``, because unintended behavior will
+/// otherwise occur when calling ``shared_from_this``.
```

Comment:
This sentence should start with `Checks that` instead of `Detects if`, because *detecting* may imply that this is the bad case, when the public inheritance is the right way to do it. (Sync with the other two locations)

Reply:
fair enough, will do.

Reply:
See https://github.com/llvm/llvm-project/pull/102299#discussion_r1787090384 for reference.